src/m_cheat.c

```diff
@@ -1306,6 +1306,9 @@ boolean M_CheatResponder(event_t *ev)
 {
   int i;
 
+  if (strictmode && demorecording)
```

Comment:
This skips every cheat, even harmless ones like `showfps`. I'd rather sacrifice that cheat than do all the processing below and also to avoid a large number of changes.

Reply:
This still keeps the chats for non-demorecording sessions. Is this intended? Was the request from dsda only addressed at demo recording?

Reply:
Should we disable all cheats (or maybe explicitly iddt) when strict mode is enabled in-game?

Reply:
DSDA-Doom's strict mode only applies when recording demos. Woof's strict mode always applies,  except for cheats after this PR. That's what OpenRift requested originally.

Here's the new behavior for Woof:

| strictmode | demorecording | allow cheats|
|---|---|---|
| no | no | yes |
| yes | no | yes |
| yes | yes | no |
| no | yes | limited |

Reply:
If you want to follow DSDA-Doom's lead, then we should abandon this PR and replace every instance of `strictmode` with something like this:

```c
// Changes to doomstat.h:

#define strictmode (current_strictmode && demorecording)

// Changes to g_game.c G_ReloadDefaults:

current_strictmode = default_strictmode;

if (M_CheckParm("-strict"))
{
  current_strictmode = true;
  force_strictmode = true;
}
```

Reply:
I don't think it's necessary to follow dsda suit here. I just want to make sure that the issue raised on Discord is properly addressed here - and I'm not sure if this was restricted to cheats during demo recording or cheats during general play.

Reply:
Got it. I double checked with kraflab and the behavior in the table is fine. So this PR is ready to go.

Reply:
Thank you for double checking!